lib/puppet/reference/configuration.rb

```diff
@@ -34,6 +34,8 @@
       val = 'Unix/Linux: /var/log/puppetlabs/puppet -- Windows: C:\ProgramData\PuppetLabs\puppet\var\log -- Non-root user: ~/.puppetlabs/var/log'
     elsif name.to_s == 'hiera_config'
       val = '$confdir/hiera.yaml. However, if a file exists at $codedir/hiera.yaml, Puppet uses that instead.'
+    elsif name.to_s == 'certname'
+      val = 'The value of `facter fqdn`'
```

Comment:
@gguillotte we probably want a bit of wordsmithing here.

Reply:
I'm not familiar with what's going on here and don't know where to look for the actual implementation, so I'm sorry if these questions have already been answered elsewhere.

First, isn't `fqdn` a [legacy fact](https://puppet.com/docs/facter/3.10/core_facts.html#fqdn)? Shouldn't it be `facter networking.fqdn`?

Secondly, are we reporting the `fqdn` fact here instead of `$trusted['certname']`, as suggested in the existing [configuration docs for the `certname` setting](https://puppet.com/docs/puppet/5.4/configuration.html#certname)?

Reply:
@gguillotte the configuration docs you linked state that the default value for certname is the node's fqnd.

From the bottom of the certname section in the doc:
"
Defaults to the node’s fully qualified domain name.

* Default: (the system’s fully qualified domain name)
"

Reply:
@McdonaldSeanp Right, but its doesn't specify that the FQDN is derived specifically from running `facter fqdn`, and also says

> A node’s certname is available in Puppet manifests as $trusted['certname'].

which is why I'm confused.

Reply:
I updated the value to be more generic:

The Host's fully qualified domain name, as determined by facter

which is a closer match to the docs anyway